src/datastore/postgres-store.ts

```diff
@@ -827,6 +836,40 @@ export class PgDataStore extends (EventEmitter as { new (): DataStoreEventEmitte
     return { found: true, result: block } as const;
   }
 
+  async getBlockByHeight(block_height: number) {
+    const result = await this.pool.query<BlockQueryResult>(
+      `
+      SELECT ${BLOCK_COLUMNS}
+      FROM blocks
+      WHERE block_height = $1
```

Comment:
This should be `WHERE block_height = $1 AND canonical = true`, otherwise it can select forked (or otherwise blocks from an invalid chainstate).